IPs/RSKIP265.md

```diff
@@ -0,0 +1,106 @@
+
+* **UTXO uneven amount distributions**: if there is a peg-in transaction P of a very high amount of bitcoins, followed of a peg-out of an amount that is higher than all other UTXO amounts, yet much lower than the amount in P, then a high amount of bitcoins will be blocked for 54 hours until the change UTXO is registered again.  If more of similar peg-out operations repeat, then a high amount of bitcoins may be locked for a long period, even if decreasing in value, blocking other peg-outs. We call this the "ladder" attack. The underlying problem can also presents itself spontaneously if the normal functioning of the peg tends to produce an uneven distribution of amounts.
+
+The solution to the denial-of-service attack is peg-out queuing batching. 
+The solution to UTXO proliferation/fragmentation is periodic, user-triggered or peg-in/peg-out triggered consolidations.
+The solution to uneven amount distribution is to rebalance amounts during consolidations.
+
+## Specification
+
+We define a series of heuristics that try to balance the number of UTXOs and distribution of amounts without generating events that consume a high percentage of the bitcoins locked, which can block the peg.
+
+### Batching
+
+The Bridge contract will wait T=3 hours to create a peg-out transaction. During the waiting time, if more peg-out requests are received, they will be queued so that all of them can be batched in a single peg-out transaction. Under normal usage, the Bridge would create no more than 8 peg-outs transactions a day. Under a rush hour of peg-outs, the Bridge may need to create more than one peg-out transaction simultaneously to reduce the transaction size. This peg-out splitting does not differ from the current protocol.
+
+### UTXO Consolidation/Expansion
+
+Every time after a peg-in transaction is registered by the Bridge, the Bridge will count the number of UTXOs and perform the following actions:
+
+* **Expansion**: If the number of UTXOs is lower than 30 and the peg-in amount is higher than 0.1 BTC, then the UTXO registered will be immediately consumed in a peg-out transaction that splits the input amount into two equally sized Powpeg outputs.
```

Comment:
... and this process is repeated till the number exceeds 30?

Reply:
No. The idea is to apply small corrections (here just 1 UTXO is added) and wait until peg-out transactions can do the rest. However, I think that using 1 transaction to create only 1 UTXO is a bad idea. I should create at least 3. I will improve this.